e2e/uimodel/ui.go

```diff
@@ -0,0 +1,45 @@
+package uimodel
+
+import (
+	"github.com/gravitational/robotest/e2e/uimodel/bandwagon"
+	"github.com/gravitational/robotest/e2e/uimodel/installer"
+	"github.com/gravitational/robotest/e2e/uimodel/opscenter"
+	"github.com/gravitational/robotest/e2e/uimodel/site"
+	"github.com/gravitational/robotest/e2e/uimodel/user"
+	web "github.com/sclevine/agouti"
+)
+
+// UI is a facade for accessing high level ui model objects
+type UI struct {
+	page *web.Page
+}
+
+// Init creates UI instance
+func Init(page *web.Page) UI {
+	return UI{page: page}
+}
+
+// GoToSite navigates to cluster page and returns cluster object
+func (u *UI) GoToSite(domainName string) site.Site {
+	return site.Open(u.page, domainName)
+}
+
+// GoToInstaller navigates to installer page and returns installer object
+func (u *UI) GoToInstaller(URL string) installer.Installer {
+	return installer.Open(u.page, URL)
+}
+
+// GoToOpsCenter navigates to opscenter page and returns opscenter object
+func (u *UI) GoToOpsCenter(URL string) opscenter.OpsCenter {
+	return opscenter.Open(u.page, URL)
+}
+
+// GoToBandwagon navigates to bandwagon page and returns bandwagon object
+func (u *UI) GoToBandwagon(domainName string) bandwagon.Bandwagon {
+	return bandwagon.Open(u.page, domainName)
+}
+
+// EnsureUser navigates to given URL and ensures that a user is logged in
+func (u *UI) EnsureUser(URL string) {
```

Comment:
1. The naming is obscure right now: I had to look inside the source of a function to understand what is it actually doing. 
2. Then I have an issue with this concept of logging in behind the scenes every time it's invoked. This approach would not allow you to catch i.e. a bug when some step user gets kicked out. Also you have this `BeforeEach()` where you could put an assert that user is logged in part of the test steps which expect user to be logged in already.
3. When you have a case as you described, when you're not sure, why not just check it and explicitly log in if needed.